src/libstd/sys/windows/ext/fs.rs

```diff
@@ -18,6 +18,43 @@ use path::Path;
 use sys;
 use sys_common::{AsInnerMut, AsInner};
 
+/// Windows-specific extensions to `File`
+#[unstable(feature = "file_offset", issue = "35918")]
+pub trait FileExt {
+    /// Seeks to a given position and reads a number of bytes.
+    ///
+    /// The offset is relative to the start of the file and thus independent
+    /// from the current cursor. The current cursor **is** affected by this
+    /// function.
+    ///
+    /// Note that similar to `File::read`, it is not an error to return with a
+    /// short read.
```

Comment:
(same as above, commenting these functions with semantics about reading/writing beyond the end of a file)